### PR TITLE
added user related APIs

### PR DESCRIPTION
**Pull request by ajay-na**

### CODE REVIEW

File: knexfile.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { config } from "dotenv";
+config();
+
+console.log(process.env.PGUSER);
```
Comment (ajay-na) [inline knexfile.ts]: remove console

File: src/app/auth/auth.controller.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+export const loginUser = async (req: Request, res: Response) => {
+  const credential: string | undefined = req.headers["x-user-auth-token"] as
+    | string
+    | undefined; // Type assertion for better type inference
+  if (credential) {
+    authService
+      .loginUser(credential)
+      .then(async (data) => {
+        const user = await authRepo.upsertUserData(
```
Comment (ajay-na) [inline src/app/auth/auth.controller.ts]: use repo in service
controller > service > repo

File: src/app/user/user.controller.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+};
+
+export const getUserById = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    const id = Number(req.params.id);
+    if (!id || typeof id != "number") {
```
Comment (ajay-na) [inline src/app/user/user.controller.ts]: move to validation function

File: src/app/user/user.controller.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      res.status(400);
+      throw new Error("Invalid id provided");
+    }
+    const data = await userService.getUserById(id);
+    if (!data) {
+      res.status(404);
+      throw new Error("User not found");
+    }
+    res.status(200).json(new SoccerQResponse(200, data, null));
```
Comment (ajay-na) [inline src/app/user/user.controller.ts]: naming can be changed

**Review by anandureghu — APPROVED** [review]

**Pull request merged**